rotkehlchen/globaldb/utils.py

```diff
@@ -51,3 +59,64 @@ def set_token_spam_protocol(
     )
     object.__setattr__(token, 'protocol', None)
     AssetResolver.clean_memory_cache(identifier=token.identifier)
+
+
+def initialize_globaldb(
+        global_dir: Path,
+        db_filename: str,
+        sql_vm_instructions_cb: int,
+) -> tuple[DBConnection, bool]:
+    """
+    Checks the database whether there are any not finished upgrades and automatically uses a
+    backup if there are any. If no backup found, throws an error.
+
+    - global_dir: The directory in which to find the global.db to initialize
+    - db_filename: The filename of the DB. Almost always: global.db.
+    - sql_vm_instructions_cb is a connection setting. Check DBConnection for details.
+
+    Returns the DB connection and true if a DB backup was used and False otherwise
+    May raise:
+        - DBUpgradeError
+    """
+    connection = DBConnection(
+        path=global_dir / db_filename,
+        connection_type=DBConnectionType.GLOBAL,
+        sql_vm_instructions_cb=sql_vm_instructions_cb,
+    )
+    try:
+        with connection.read_ctx() as cursor:
+            ongoing_upgrade_from_version = globaldb_get_setting_value(
+                cursor=cursor,
+                name='ongoing_upgrade_from_version',
+                default_value=-1,
+            )
+    except sqlite3.OperationalError:  # pylint: disable=no-member
+        ongoing_upgrade_from_version = -1  # Fresh DB
+
+    if ongoing_upgrade_from_version == -1:
+        return connection, False  # We are all good
+
+    # Otherwise replace the db with a backup and relogin
+    connection.close()
```

Comment:
do we have a test for this code? I can easily see the splitting of this function breaking something.

Reply:
every test in the codebase that uses the globaldb calls this method.